Use the new product page everywhere

jasmeet0817 opened the pull request:
Removed the old product_page.dart and renamed new_product_page.dart -> product_page.dart

Commit:
Use the new product page everywhere

Commit:
Merge

Commit:
Merge

Commit:
mformat

monsieurtanuki reviewed (changes requested):
Hi @jasmeet0817!
I would have kept `NewProductPage` as a file in order to keep its code history, and just branch everything to it replacing the calls to the (old) `ProductPage`. Because in this PR unfortunately:
1. the reviewers cannot say what you did in `ProductPage` - as you changed everything
2. you're still keeping a copy of the code in the `NewProductPage` file that you didn't remove
3. you didn't remove either all the related code that was only called by the old product page

My suggestion: branch everything to `NewProductPage`, and that's it.
I can do the code removal around (old)`ProductPage` afterwards.
Does that sound ok?

Commit:
...

Commit:
...

Commit:
...

monsieurtanuki commented on packages/smooth_app/lib/pages/product/product_page.dart:
```diff
@@ -76,11 +76,11 @@ class _ProductPageState extends State<ProductPage> {
         Future<void>.delayed(
           Duration.zero,
-          () => ScaffoldMessenger.of(context).showSnackBar(
+              () => ScaffoldMessenger.of(context).showSnackBar(
```
strange tab

monsieurtanuki reviewed (approved):
I'm old-fashioned and a bit puzzled by `new_product_page.dart` hosting `ProductPage`, but that's ok in dart/flutter so why not.
Strange tabs here and there, but that's ok too I guess.
Good job @jasmeet0817!

Commit:
format

Commit:
...

jasmeet0817 commented:
> Hi @jasmeet0817! I would have kept `NewProductPage` as a file in order to keep its code history, and just branch everything to it replacing the calls to the (old) `ProductPage`. Because in this PR unfortunately:
> 
> 1. the reviewers cannot say what you did in `ProductPage` - as you changed everything

Fair, I brought back new_product_page file, but I sti
> 2. you're still keeping a copy of the code in the `NewProductPage` file that you didn't remove
> 3. you didn't remove either all the related code that was only called by the old product page
> 
> My suggestion: branch everything to `NewProductPage`, and that's it. I can do the code removal around (old)`ProductPage` afterwards. Does that sound ok?



> I'm old-fashioned and a bit puzzled by `new_product_page.dart` hosting `ProductPage`, but that's ok in dart/flutter so why not. Strange tabs here and there, but that's ok too I guess. Good job @jasmeet0817!

Thank you, good points. Fixed most issues. Renamed to `ProductPage` because I think there's no concept of `New` anymore. Once you delete the old code, it would make sense.

Commit:
Remove the link to new product page from the old one (which is now dereferrenced)
